src/flashbar/index.tsx

```diff
@@ -1,85 +1,178 @@
-import { useReducedMotion, useVisualRefresh } from '../internal/hooks/use-visual-mode';
+import customCssProps from '../internal/generated/custom-css-properties';
+import { Flash } from './flash';
 import { FlashbarProps } from './interfaces';
+import { getBaseProps } from '../internal/base-component';
+import InternalIcon from '../icon/internal';
+import { TIMEOUT_FOR_ENTERING_ANIMATION } from './constant';
 import { TransitionGroup } from 'react-transition-group';
 import { Transition } from '../internal/components/transition';
-import { TIMEOUT_FOR_ENTERING_ANIMATION } from './constant';
+import useBaseComponent from '../internal/hooks/use-base-component';
+import { useContainerBreakpoints } from '../internal/hooks/container-queries';
+import useFocusVisible from '../internal/hooks/focus-visible';
+import { useMergeRefs } from '../internal/hooks/use-merge-refs';
+import { useReducedMotion, useVisualRefresh } from '../internal/hooks/use-visual-mode';
+import { getVisualContextClassname } from '../internal/components/visual-context';
+import styles from './styles.css.js';
 
 export { FlashbarProps };
 
 export default function Flashbar({ items, ...restProps }: FlashbarProps) {
   const { __internalRootRef } = useBaseComponent('Flashbar');
   const [breakpoint, ref] = useContainerBreakpoints(['xs']);
-  const isRefresh = useVisualRefresh();
   const baseProps = getBaseProps(restProps);
-
   const mergedRef = useMergeRefs(ref, __internalRootRef);
+  const isFocusVisible = useFocusVisible();
+  const isVisualRefresh = useVisualRefresh();
+
+  /**
+   * All the flash items should have ids so we can identify which DOM element is being
+   * removed from the DOM to animate it. Motion will be disabled if any of the provided
+   * flash messages does not contain an `id`.
+   */
+  const motionDisabled =
+    useReducedMotion(ref as any) || !isVisualRefresh || (items && !items.every(item => 'id' in item));
+
+  /**
+   * The `enableStackingOption` property is a hidden boolean that allows for teams
+   * to beta test the flashbar stacking feature.
+   */
+  const enableStackingOption = (restProps as any).enableStackingOption;
+  const [isFlashbarStacked, setIsFlashbarStacked] = useState(false);
+  const [isFlashbarStackExpanded, setIsFlashbarStackExpanded] = useState(false);
 
-  // All the flash items should have ids so we can identify which DOM element is being removed from the DOM to animate it.
-  const motionDisabled = useReducedMotion(ref as any) || !isRefresh || (items && !items.every(item => 'id' in item));
-  // Motion will be disabled if any of the provided flash messages does not contain an `id`
-
-  const renderFlashItem = (item: FlashbarProps.MessageDefinition, index: number) => (
-    <Flash
-      key={item.id ?? index}
-      // eslint-disable-next-line react/forbid-component-props
-      className={clsx(isRefresh ? styles['flash-refresh'] : '')}
-      {...item}
-    />
+  useEffect(
```

Comment:
This should not be an effect: https://beta.reactjs.org/learn/you-might-not-need-an-effect#updating-state-based-on-props-or-state